### PR TITLE
fix: major exponent

### PR DESCRIPTION
**Pull request by dolcalmi**
Update minorToMajorUnit and majorToMinorUnit to use currency major exponent

Merge after https://github.com/GaloyMoney/galoy/pull/2439

### CODE REVIEW

File: src/app/cold-storage/rebalance-to-cold-wallet.ts [hunk 1]
```diff
@@ -81,19 +81,21 @@ export const rebalanceToColdWallet = async (): Promise<boolean | ApplicationErro
   })
   if (rebalanceBtcAmount instanceof Error) return rebalanceBtcAmount
   const amountDisplayCurrencyAmount = displayPriceRatio.convertFromBtc(rebalanceBtcAmount)
-  const amountDisplayCurrency = usdMinorToMajorUnit(
-    amountDisplayCurrencyAmount.amount,
-  ) as DisplayCurrencyBaseAmount
+  const amountDisplayCurrency = minorToMajorUnit({
```
Comment (vindard) [inline src/app/cold-storage/rebalance-to-cold-wallet.ts:84]: I'm noticing here now that there's some cleanup I need to do with calling WalletPriceRatio/DisplayPriceRatio still. I'll probably start building it on these PRs

File: src/app/wallets/update-on-chain-receipt.ts [hunk 2]
```diff
@@ -249,7 +244,7 @@ const processTxForWallet = async (
             amount: Number(
               minorToMajorUnit({
                 amount: creditAccountAdditionalMetadata.displayAmount,
-                displayMajorExponent: MajorExponent.STANDARD,
+                displayCurrency: creditAccountAdditionalMetadata.displayCurrency,
```
Comment (vindard) [inline src/app/wallets/update-on-chain-receipt.ts:247]: Like here. I think after my refactor (after this branch) the `displayCurrency` could come directly from `amount` field if it's typed as `DisplayAmount` and then we could remove the `displayCurrency` param here

File: src/domain/fiat/display-currency.ts [hunk 3]
```diff
@@ -5,40 +5,44 @@ export const MajorExponent = {
...
   amount,
   displayCurrency,
 }: {
   amount: number | bigint
   displayCurrency: DisplayCurrency
-}) => {
+}): number => {
   const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
-  return majorToMinorUnit({ amount, displayMajorExponent })
+  return Number(Number(amount) * 10 ** displayMajorExponent)
```
Comment (vindard) [inline src/domain/fiat/display-currency.ts:45]: Could `Number(amount) * 10 ** displayMajorExponent`

File: src/domain/fiat/display-currency.ts [hunk 4]
```diff
@@ -5,40 +5,44 @@ export const MajorExponent = {
...
-
-// TODO: remove after remove hardcoded DisplayCurrency.Usd
-export const usdMajorToMinorUnit = (amount: number | bigint) =>
-  majorToMinorUnit({ amount, displayMajorExponent: MajorExponent.STANDARD })
+  displayCurrency: DisplayCurrency
+  fixed?: boolean
+}): number => {
+  const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
+  const majorAmount = Number(amount) / 10 ** displayMajorExponent
+  return fixed ? Number(majorAmount.toFixed(displayMajorExponent)) : majorAmount
```
Comment (vindard) [inline src/domain/fiat/display-currency.ts:34]: I think the two values in this conditional work out to the same, redundant conditional maybe. Looks like the string stuff is covered with `minorToMajorUnitFormatted` anyway too.

**Review by vindard — COMMENTED** [review]
Left a few small comments

File: src/domain/fiat/display-currency.ts (hunk 4 above)
Comment (dolcalmi) [reply src/domain/fiat/display-currency.ts:34]: `fixed: false` it is needed for displayCurrencyPerSettlementCurrencyMajorUnit

File: src/domain/fiat/display-currency.ts (hunk 3 above)
Comment (dolcalmi) [reply src/domain/fiat/display-currency.ts]: Fixed https://github.com/GaloyMoney/galoy/pull/2450/commits/d0583cab39d5ca7c0c4ebe6daf5d4ab0c2f529e6

File: src/domain/fiat/display-currency.ts (hunk 4 above)
Comment (vindard) [reply src/domain/fiat/display-currency.ts:34]: Oh right, the `fixed` name isn't intuitive though. ChatGPT recommended the following:
- [ ] useRounded
- [ ] applyPrecision
- [ ] roundResult
- [x] exactPrecision
- [ ] isRounded

I like the `exactPrecision` one

**Review by vindard — CHANGES_REQUESTED** [review]
Just one requested change, but otherwise this look good

**Review by vindard — APPROVED** [review]

**Pull request merged**